### PR TITLE
Update to get latest release from S3

### PR DESCRIPTION
**Pull request by jaydeland**
Updates to support mmctl being published to S3

Paths will need to be updated

### CODE REVIEW

**Review by jaydeland — COMMENTED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -188,14 +188,9 @@ prepackaged-plugins: ## Populate the prepackaged-plugins directory
...
-	@echo "mmctl already exists in bin/mmctl not downloading a new version."
+	@echo "MMCTL already exists in bin/mmctl not downloading a new version."
 else
-	@MMCTL_VERSION=$$(scripts/get_latest_release.sh mattermost/mmctl release-); if [ $$? -eq 0 ]; then \
-		scripts/download_mmctl_release.sh $$MMCTL_VERSION; \
-	else \
-		echo $$MMCTL_VERSION; \
-		exit 1; \
-	fi;
+	@scripts/download_mmctl_release.sh;
```
Comment (khos2ow) [inline Makefile:193]: `;` at the end of the line is technically not needed.

File: build/release.mk [hunk 2]
```diff
@@ -97,8 +97,8 @@ package:
 	mkdir -p $(DIST_PATH)/client
 	cp -RL $(BUILD_WEBAPP_DIR)/dist/* $(DIST_PATH)/client
 
-	$(eval MMCTL_REL_TO_DOWNLOAD := $(shell scripts/get_latest_release.sh 'mattermost/mmctl' 'release-'))
-	@echo "Using mmctl version $(MMCTL_REL_TO_DOWNLOAD)"
+  #Download MMCTL
```
Comment (khos2ow) [inline build/release.mk:100]: looks like a missed indentation with `SPACE` instead of `TAB`

File: build/release.mk [hunk 3]
```diff
@@ -97,8 +97,8 @@ package:
...
+	$(shell scripts/get_latest_release.sh)
```
Comment (khos2ow) [inline build/release.mk:101]: technically `$(shell ...)` is not needed and `scripts/get_latest_release.sh` can be called directly.

File: scripts/download_mmctl_release.sh [hunk 4]
```diff
@@ -9,21 +9,28 @@ else
 fi
 
 # strip whitespace
-RELEASE_TO_DOWNLOAD=$(echo "$1" | xargs echo)
-echo "Downloading prepackaged binary: https://github.com/mattermost/mmctl/releases/$RELEASE_TO_DOWNLOAD";
+THIS_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+if [[ $THIS_BRANCH =~ 'release-' ]];
+then
+  RELEASE_TO_DOWNLOAD=$THIS_BRANCH
```
Comment (khos2ow) [inline scripts/download_mmctl_release.sh:15]: It's better to wrap `$THIS_BRANCH` in `" "` to prevent globbing and expansion.

**Review by khos2ow — COMMENTED** [review]
lgtm, some nitpicks below.

**Review by khos2ow — APPROVED** [review]

**Pull request merged**